### ISSUE TITLE
Creating multiple OpenAPIBackend classes with shared default handlers cause data corruption

### ISSUE THREAD

**Issue by henhal** [anttiviljami/openapi-backend#112]
This is something I hit which took me a while to figure out.
I was creating a server that implemented two different API definitions, so I created two instances of `OpenAPIBackend`. Each one was given its own definition and its own set of operation handlers.
However, in the constructor I also submitted a set of default handlers for `notFound` and such. This object was passed to both instances of `OpenAPIBackend`.

In the constructor, there's this line to copy the options passed:

```
const optsWithDefaults = Object.assign({ apiRoot: '/', validate: true, strict: false, quick: false, ajvOpts: {}, handlers: {}, 
    securityHandlers: {} }, opts);
```

While this creates a new `opts` object, it does not deep-copy `opts`, so `opts.handlers` and `opts.securityHandlers` are copied by reference. This can result in weird behaviour if a constant is passed around, ending up being a shared mutable property on the OpenAPIBackend objects.

The result of this was that each instance used the same object for the map between operationIds and handlers, meaning a lot of warnings about unknown operation etc, since `OpenAPIBackend` 1 had a definition with an operation `getFoo` but operations `getFoo` and `getBar` registered in the handlers object, and `OpenAPIBackend` 2 had a definition with an operation `getBar` but also both operations in the handlers object.

Simple PoC:

test.ts:

```
import * as OpenAPI from 'openapi-backend';

const handlers = {
    notFound() {
        throw new Error('not found');
    },
};

async function test() {
    const api1 = new OpenAPI.OpenAPIBackend({
        definition: __dirname + '/foo.yml',
        apiRoot: '/foo',
        handlers
    });

    api1.registerHandler('getFoo', () => {});
    await api1.init();

    console.log('after registering api1');
    console.log('api1:', api1);

    const api2 = new OpenAPI.OpenAPIBackend({
        definition: __dirname + '/bar.yml',
        apiRoot: '/bar',
        handlers
    });

    api2.registerHandler('getBar', () => {});
    await api2.init();

    console.log('after registering api2');

    console.log('api1:', api1);
    console.log('api2:', api2);
}

test();
```

foo.yml:

```
openapi: 3.0.2
info:
  title: 'FOO'
  version: 1.0.0
paths:
  /foo:
    get:
      operationId: getFoo
      responses:
        '200':
          description: OK
          content:
            application/json:
              schema:
                type: object
                properties:
                  message:
                    type: string
```

bar.yml:

```
openapi: 3.0.2
info:
  title: 'BAR'
  version: 1.0.0
paths:
  /bar:
    get:
      operationId: getBar
      responses:
        '200':
          description: OK
          content:
            application/json:
              schema:
                type: object
                properties:
                  message:
                    type: string
```

If adding `Object.freeze()` around my supposedly constant `handlers` object, I got an error instead, obviously, which puts a light on the problem (and the fact that one should really use `Object.freeze` more :) ). Another workaround is to pass `{...handlers}` instead of `handlers` to the constructor.

However, I'm wondering if you have any thoughts on whether this is really reasonable behaviour, or if the constructor should in fact _copy_ the `handlers` and `securityHandlers` objects to prevent things like this? I think it's fairly often a good idea to deep copy options that are passed, but it's not _always_ feasible so I won't say that it's definitely the right thing. But since `opts.handlers` and `opts.securityHandlers` are maps which might very well be re-used from constants, I think it wouldn't hurt copying them?

**Comment by henhal** [comment]
(I have no problem whipping up a PR for this if you agree it would be an improvement and there are no side-effects.)

**Comment by anttiviljami** [comment]
Excellent investigation work! PR would be very much appreciated 😍

**Comment by anttiviljami** [comment]
Fixed in release `3.5.4`
